### PR TITLE
[build] add MADVDONTNEED to builds

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## What does this PR do?

This adds `GODEBUG="madvdontneed=1"` to the docker images and init scripts.
Note that I haven't tested this yet as I'm not sure what to check beyond "the env var exists in the final builds"

## Why is it important?

See https://github.com/elastic/beats/issues/22786 From the linked issue:

> The go runtime defaults to MADV_FREE by default. In that case pages are still assigned with the Beat, even after the pages have been "returned" to the OS. The kernel will eventually claim these pages, but RSS memory usage is reported as very high (which it is not necessarily). This seems to mess with the OOM killer at times.

## Checklist

- [X] My code follows the style guidelines of this project
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- https://github.com/elastic/beats/issues/22786

### CODE REVIEW

**Comment by fearful-symmetry** [comment]
So, we're building into the docker images:
```
            "Env": [
                "PATH=/usr/share/metricbeat:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "ELASTIC_CONTAINER=true",
                "GODEBUG=madvdontneed=1"
            ],
```

File: dev-tools/packaging/templates/deb/elastic-agent.sh.tmpl [hunk 1]
```diff
@@ -23,6 +23,8 @@ WRAPPER="/usr/share/${NAME}/bin/${NAME}-god"
 BEAT_USER="{{.BeatUser}}"
 WRAPPER_ARGS="-r / -n -p $PIDFILE"
 SCRIPTNAME=/etc/init.d/{{.ServiceName}}
+DEFAULT_GODEBUG="madvdontneed=1"
+export GODEBUG=${GODEBUG-$DEFAULT_GODEBUG}
```
Comment (jsoriano) [inline dev-tools/packaging/templates/deb/elastic-agent.sh.tmpl:27]: Usually, environment overrides are done in `/etc/default` files, so this should be done after reading `/etc/default/{{.ServiceName}}` some lines below.

File: dev-tools/packaging/templates/deb/init.sh.tmpl [hunk 2]
```diff
@@ -24,6 +24,8 @@ WRAPPER="/usr/share/${NAME}/bin/${NAME}-god"
 BEAT_USER="{{.BeatUser}}"
 WRAPPER_ARGS="-r / -n -p $PIDFILE"
 SCRIPTNAME=/etc/init.d/{{.ServiceName}}
+DEFAULT_GODEBUG="madvdontneed=1"
+export GODEBUG=${GODEBUG-$DEFAULT_GODEBUG}
```
Comment (jsoriano) [inline dev-tools/packaging/templates/deb/init.sh.tmpl:28]: Also this line, please move it after reading `/etc/default/{{.ServiceName}}`.

**Review by jsoriano — COMMENTED** [review]
Looks good, thanks for adding this! Only a comment about the defaults in debian init files.

**Comment by fearful-symmetry** [comment]
@jsoriano thanks, fixed!

**Review by jsoriano — APPROVED** [review]
Thanks!

**Review by urso — APPROVED** [review]

**Pull request merged**